packages/analytics/amplify_analytics_pinpoint/lib/src/legacy_native_data_provider/flutter_legacy_native_data_provider.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+import 'dart:io' show Platform;
+
+import 'package:amplify_analytics_pinpoint/src/legacy_native_data_provider/data_provider.android.dart';
+import 'package:amplify_analytics_pinpoint/src/legacy_native_data_provider/data_provider.ios.dart';
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+
+/// {@template amplify_analytics_pinpoint.flutter_legacy_native_data_provider}
+/// Provides legacy data stored by Amplify Analytics iOS and Android.
+/// {@endtemplate}
+class FlutterLegacyNativeDataProvider implements LegacyNativeDataProvider {
+  /// {@macro amplify_analytics_pinpoint.flutter_legacy_native_data_provider}
+  factory FlutterLegacyNativeDataProvider() {
+    LegacyNativeDataProvider? provider;
+
+    if (Platform.isIOS) {
```

Comment:
This line crashed on web for me.

suggestion: `if (!zIsWeb && Platform.isIOS) `